api/src/main/java/org/pmiops/workbench/db/model/User.java

```diff
@@ -107,4 +117,21 @@ public String getFreeTierBillingProjectName() {
   public void setFreeTierBillingProjectName(String freeTierBillingProjectName) {
     this.freeTierBillingProjectName = freeTierBillingProjectName;
   }
+
+  // Authorities (special permissions) are granted only through direct db edits, for example
+  // INSERT INTO authority VALUES(
+  //     (SELECT user_id FROM user WHERE email = 'me@staging.pmi-ops.org'),
+  //     0);
+  // TODO(RW-85) Tool to edit authorities.
+  // TODO Switch to FetchMode.JOIN when if upgrade past Hibernate 3 (EAGER is deprecated).
```

Comment:
After reading these posts, it sounds like @Fetch(FetchMode.JOIN) may not work with JPA:

https://stackoverflow.com/questions/463349/jpa-eager-fetch-does-not-join
https://stackoverflow.com/questions/29602386/how-does-the-fetchmode-work-in-spring-data-jpa

I don't love adding an extra database lookup to every request we handle to look up authorities, when only a few, rarely accessed endpoints will be annotated as requiring an authority. 

If you make this a lazy fetch, and only look up the authorities for the user once you've determined that the method requires some authorities, this won't be a problem.

Reply:
Understanding how the ORM you're working works (and doesn't work) is the cost of every ORM system. All ORMs have their limitations. Not being able to specify an eager join easily is annoying, I agree -- Hibernate handles this somewhat better than JPA. (On the other hand, Hibernate involves more boilerplate and setup.)

The fact that we haven't had to write any actual SQL code or tests for it yet in workbench -- our DAOs are just interfaces -- shows that JPA is working quite well for us. It's taking care of all the SQL and transformations of object fields -- what would be a few hundred lines of code already, and soon to be more. Not having to write or maintain that code is absolutely a win.

In any case: the default (lazy fetching) is fine in this particular case -- doing a join isn't a good idea when we very rarely need this data. You can remove the TODO here.